raster/src/main/scala/geotrellis/raster/io/geotiff/writer/TiffTagFieldValue.scala

```diff
@@ -187,7 +188,15 @@ object TiffTagFieldValue {
     // Tags that are different if it is striped or tiled storage, and a function
     // that sets up a tag to point to the offsets of the image data.
 
-    val segmentByteCounts = imageData.segmentBytes.map { _.length }.toArray
+    val segmentByteCounts = {
+      val len = imageData.segmentBytes.length
+      val arr = Array.ofDim[Int](len)
+      cfor(0)(_ < len, _ + 1) { i =>
+        arr(i) = imageData.segmentBytes.getSegmentByteCount(i)
+      }
+      arr
+    }
```

Comment:
@echeipesh @jbouffard this is the offending line for the streaming problem. Notice the doc string I made for SegmentBytes. The fixed line's reliance on the ordering of the segments must have been missed when SegmentBytes moved from always being segment order (which was clear since it used to only be `Array[Array[Byte]]` to an arbitrary order based on the `getSegments(0 until length)` call.